### PR TITLE
Eco 1312/tx history char limits

### PR DESCRIPTION
**Pull request by OrenZak**
#### Main purpose:
Set text limitations EM based.
#### Technical description:
- Separated the title text and set limit 
- Separated the subtitle text and set limit
- Updated sample app with longer text for native offers

### CODE REVIEW

File: sdk/src/main/java/com/kin/ecosystem/history/view/OrderHistoryRecyclerAdapter.java [hunk 1]
```diff
@@ -125,46 +125,47 @@ private void setAmount(Order item) {
         }
 
         private void setSubtitle(Order item) {
-            StringBuilder subTitle = new StringBuilder();
+            String subTitle = "";
```
Comment (shayb) [inline sdk/src/main/java/com/kin/ecosystem/history/view/OrderHistoryRecyclerAdapter.java:130]: why removed the string builder?

File: sdk/src/main/java/com/kin/ecosystem/history/view/OrderHistoryRecyclerAdapter.java [hunk 2]
```diff
@@ -125,46 +125,47 @@ private void setAmount(Order item) {
...
-            setText(R.id.action_text, actionText);
-            if(item.getStatus() == Status.FAILED) {
-                setText(R.id.title, title + delimiter);
+            if(isOrderFailed(item.getStatus())) {
+                setText(R.id.title, title);
+                setMaxEMs(R.id.title, 9);
+                setVisibility(R.id.delimiter, VISIBLE);
             } else  {
                 setText(R.id.title, title);
+                setMaxEMs(R.id.title, 12);
```
Comment (shayb) [inline sdk/src/main/java/com/kin/ecosystem/history/view/OrderHistoryRecyclerAdapter.java]: const?

File: sdk/src/main/java/com/kin/ecosystem/history/view/OrderHistoryRecyclerAdapter.java [hunk 3]
```diff
@@ -125,46 +125,47 @@ private void setAmount(Order item) {
...
+                setVisibility(R.id.delimiter, GONE);
             }
+            setActionText(item);
         }
 
-        private boolean isFailed(Status status) {
+        private boolean isOrderFailed(Status status) {
             return status == Status.FAILED;
```
Comment (shayb) [inline sdk/src/main/java/com/kin/ecosystem/history/view/OrderHistoryRecyclerAdapter.java:158]: can put the method in the enum it selft

File: sdk/src/main/res/layout/kinecosystem_offer_recycler_item.xml [hunk 4]
```diff
@@ -27,9 +27,9 @@
         android:layout_width="wrap_content"
         android:layout_height="wrap_content"
         android:layout_marginLeft="@dimen/kinrecovery_margin_large"
-        android:ellipsize="end"
+        android:singleLine="true"
+        android:ellipsize="marquee"
```
Comment (shayb) [inline sdk/src/main/res/layout/kinecosystem_offer_recycler_item.xml:31]: amazing

File: sdk/src/main/java/com/kin/ecosystem/history/view/OrderHistoryRecyclerAdapter.java (hunk 1 above)
Comment (OrenZak) [reply sdk/src/main/java/com/kin/ecosystem/history/view/OrderHistoryRecyclerAdapter.java:130]: it's separated now for the description view and date view

File: sdk/src/main/java/com/kin/ecosystem/history/view/OrderHistoryRecyclerAdapter.java (hunk 3 above)
Comment (OrenZak) [reply sdk/src/main/java/com/kin/ecosystem/history/view/OrderHistoryRecyclerAdapter.java:158]: it's a generated file so, i don't want to change anything only if i must.

**Pull request merged**